packages/amplify-e2e-tests/src/__tests__/auth_2a.test.ts

```diff
@@ -47,5 +52,12 @@ describe('amplify add auth...', () => {
     expect(clients[0].UserPoolClient.CallbackURLs[0]).toEqual('https://www.google.com/');
     expect(clients[0].UserPoolClient.LogoutURLs[0]).toEqual('https://www.nytimes.com/');
     expect(clients[0].UserPoolClient.SupportedIdentityProviders).toHaveLength(5);
+
+    // amplify pull should work
+    const functionName = `testcorsfunction${generateRandomShortId()}`;
+    const projRoot2 = await createNewProjectDir('auth2');
+    await addFunction(projRoot, { functionTemplate: 'Hello World', name: functionName }, 'nodejs');
+    await amplifyPull(projRoot2, { emptyDir: true, appId, envName: 'integtest', yesFlag: true });
+    deleteProjectDir(projRoot2);
```

Comment:
If `addFunction()` or `amplifyPull()` fails, this deletion won't be called. Can the folder creation/deletion for `projRoot2` be moved to the `beforeEach` and `afterEach` events?